moveit_core/collision_detection/src/pycollision_detection.cpp

```diff
@@ -0,0 +1,67 @@
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/eigen.h>
+#include <moveit/collision_detection/world.h>
+#include <moveit/collision_detection/collision_matrix.h>
+#include <moveit/collision_detection/collision_common.h>
+#include <moveit/python/pybind_rosmsg_typecasters.h>
+
+namespace py = pybind11;
+
+using namespace collision_detection;
+
+auto SET_ENTRY1 = py::overload_cast<const std::string&, const std::string&, bool>(&AllowedCollisionMatrix::setEntry);
+
+void def_collision_detect_bindings(py::module& m)
+{
+  m.doc() = "World represents the state of the collision objects, robot, and other environment information";
+  py::class_<World, WorldPtr>(m, "World").def(py::init<>());
+  py::class_<CollisionRequest>(m, "CollisionRequest")
+      .def(py::init<>())
+      .def_readwrite("contacts", &CollisionRequest::contacts)
```

Comment:
For a future version of pybind11, you will need to define `Contact` first. Please reorder your `class_` definitions accordingly.

Reply:
Ok I think I've address this